nav2_behavior_tree/plugins/condition/is_new_goal_condition.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__IS_NEW_GOAL_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__IS_NEW_GOAL_CONDITION_HPP_
+
+#include <string>
+
+#include "behaviortree_cpp_v3/condition_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class IsNewGoalCondition : public BT::ConditionNode
+{
+public:
+  IsNewGoalCondition(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf)
+  : BT::ConditionNode(condition_name, conf)
+  {
+    config().blackboard->set("new_goal_received", false);
+  }
+
+  IsNewGoalCondition() = delete;
+
+  BT::NodeStatus tick() override
+  {
+    // Check if there's a new goal on the blackboard
+    if (config().blackboard->get<bool>("new_goal_received")) {
+      // Reset the flag in the blackboard
+      config().blackboard->set("new_goal_received", false);
```

Comment:
Do not set false here, that will effect the state of potentially other BT nodes that need that flag (e.g. multiple instances of this)